srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/PopulatedDisplayGroup.java

```diff
@@ -0,0 +1,204 @@
+            }
+        }
+        
+        // Preconditions satisfied: build views of all of the necessary variables
+        // so that they can be represented as DisplayItem instances.
+        // Store variables for reference later.
+        fVariables = new ArrayList<>(variables);
+        fDisplayItems = new ArrayList<>();
+        for(final Variable var: variables)
+        {
+            // TODO: If the variable has specified keys that have reference information,
+            // create a String that holds that reference information and use it in the view constructors.
+            VariableView varView;
+            if(var instanceof BooleanVariable)
+            {
+                varView = new BooleanVariableView((BooleanVariable) var, "");
+            }
+            else if(var instanceof DiscreteNumericalVariable)
+            {
+                varView = new DiscreteNumericalVariableView((DiscreteNumericalVariable) var, "");
+            }
+            else if(var instanceof NumericalVariable)
+            {
+                varView = new NumericalVariableView((NumericalVariable) var, "");
+            }
+            else if(var instanceof MultiSelectVariable)
+            {
+                varView = new MultiSelectVariableView((MultiSelectVariable) var, "");
+            }
+            else if(var instanceof ProcedureVariable)
+            {
+                varView = new ProcedureVariableView((ProcedureVariable) var, "");
+            }
+            else
+            {
+                LOGGER.debug("Unexpected Variable type while creating a PopulatedDisplayGroup. Key: {}",
+                        var.getKey());
+                // Unexpected variable type
+                continue;
+            }
+            fDisplayItems.add(varView);
+        }
+        
+        // These group names are all well-known and are hard-coded as such.
+        if(group.getName().equalsIgnoreCase("Clinical Conditions or Diseases - Recent"))
+        {
+            final List<String> factorList = new ArrayList<String>();
+            // Output the date without the time of day.
+            final SimpleDateFormat format = new SimpleDateFormat("MM/dd/yy");
+            for(final HealthFactor factor: patient.getHealthFactors())
+            {
+                factorList.add(String.format("%s %s%n", format.format(factor.getDate()), factor.getFactor()));
+            }
+            final DisplayItem refInfo = new ReferenceItem("Health Factors", variables.get(0).getGroup(), factorList);
+            fDisplayItems.add(0, refInfo);
+        }
+    }
+
+    /**
+     * <p>Returns the List of Variables, always containing at least 1 Variable. All
+     * contained Variables will have equal {@link VariableGroup}s. The
+     * collection itself is immutable, though the variables themselves are not.</p>
+     * 
+     * <p>Order is preserved from construction.</p>
+     */
+    public List<Variable> getVariables()
```

Comment:
Is this accessor necessary? I don't see why we need to store the original Variable list.